pallets/xyk/src/mock.rs

```diff
@@ -56,18 +57,34 @@ impl system::Trait for Test {
     type SystemWeightInfo = ();
 }
 
-impl Trait for Test {
+impl orml_tokens::Trait for Test {
     type Event = ();
+    type Balance = u128;
+    type Amount = i128;
+    type CurrencyId = u32;
+    type OnReceived = ();
+    type WeightInfo = ();
 }
 
-impl assets::Trait for Test {
-    type Balance = u128;
-    type AssetId = u32;
+impl Trait for Test {
     type Event = ();
+    type Currency = MultiTokenCurrencyAdapter<Test>;
 }
 
 pub type XykStorage = Module<Test>;
 
+impl<T: Trait> Module<T> {
```

Comment:
why is this required?

Reply:
just to keep test code a bit simpler/readable without need to cast type N times in single test case scenario